frontend/src/metabase-lib/lib/queries/structured/Join.js

```diff
@@ -406,23 +406,59 @@ export default class Join extends MBQLObjectClause {
     }
   }
 
-  setJoinDimension({ index = 0, dimension }) {
+  setJoinDimension({ index = 0, dimension, overwriteTemporalUnit = false }) {
     const condition = this.getConditionByIndex(index);
-    const newCondition = [
-      "=",
-      condition ? condition[PARENT_DIMENSION_INDEX] : null,
-      this._convertDimensionIntoMBQL(dimension),
-    ];
+    let joinDimension = this._convertDimensionIntoMBQL(dimension);
+    let parentDimension = condition ? condition[PARENT_DIMENSION_INDEX] : null;
+
+    if (
```

Comment:
I'd prefer if we pulled this new `if` block out of `setJoinDimension` and `setParentDimension` to keep them simpler. I think something like the following could be nice:
```
const newCondition = isDateTimeFieldJoin(...)
  ? getDateTimeFieldCondition(...)
  : ["=", parentDimension, joinDimension]
```
But feel free to abstract in whatever way makes sense to you